### ISSUE TITLE
New splice API issue with paths containing '/'

### ISSUE THREAD

**Issue by cgauld** [automerge/automerge#825]
We're investigating the use of the new `splice` function since we're having some issues with handling unicode and how we can reconcile changes (in both directions) between an Automerge doc and the Monaco text editor.

We've encountered an issue where `splice` errors with the following exception:
```
 RangeError: Cannot splice: RangeError: invalid op for object of type `map`
```
We suspect it may be due to our path containing a key with a '/' in it. Our doc structure looks like this:
```ts
{
   "unsaved": {
        "/index.ts": {
            "txt: "..."
        }
    }
}
```
It looks like the `value` function inside `splice` ends up taking the following value:
```
_root/unsaved//index.ts/txt
```
Which seems like it may be problematic. Would be great to get your thoughts :-)

**Comment by acurrieclark** [comment]
That sounds about right, unfortunately. The path array is joined with forwardslashes before being passed on.

as a workaround, rather than passing in your root document and then the path all the way to the spliceable element of text, you can pass in the parent object containing the text itself and then use the key as the path

**Comment by acurrieclark** [comment]
No problem! Glad to help.

@alexjg worth keeping this open for you for any reason?

**Comment by alexjg** [comment]
Yup, I'm keeping this open because I think  we should support paths with slashes in them. This will require a bit of surgery though as we'll need to change the `automerge-wasm` API and check that it performs acceptably.
